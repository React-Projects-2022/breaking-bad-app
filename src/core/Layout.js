import { Routes, Route } from "react-router-dom";
import { PATHS } from "./../helpers/paths";
import Footer from "./Footer";
import Episodes from "./../pages/Episodes";
import Header from "./Header";
import Home from "./../pages/Home";
import PrincipalCharsVotes from "../pages/PrincipalCharsVotes";
const NoMatch = () => <div>No match</div>

const Layout = () => {
  return (
    <>
      <Header />
      <div className="container-md">
        <Routes>
          <Route path={PATHS.HOME} element={<Home />} />
          <Route path={PATHS.EPISODES} element={<Episodes />} />
          <Route path={PATHS.VOTES} element={<PrincipalCharsVotes />} />
          <NoMatch default />
        </Routes>
      </div>
      <Footer />
    </>
  );
};

export default Layout;
