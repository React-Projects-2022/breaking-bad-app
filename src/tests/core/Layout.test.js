import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Layout from "../../core/Layout";


describe("Pruebas en Layout", () => {
  const wrapper = shallow(<Layout />);

  test("Debe de mostrar <Layout /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobar que Header y Footer son añadidos correctamente", () => {
    expect(wrapper.find('Header').exists()).toBe(true);
    expect(wrapper.find('Footer').exists()).toBe(true)
  });

  test("Comprobar que elemento Routes y sus Route inidividuales están OK", () => {
    expect(wrapper.find('Routes').exists()).toBe(true);
    expect(wrapper.find('Route').length).toBe(3);
  });

  test("Comprobando paths de los 'Route'", () => {
    const labels = ["", "/episodes", "/votes"];
    labels.map((element, index) =>
      expect(wrapper.find("Route").at(index).prop('path')).toBe(`/breaking-bad-app${element}`)
    );
  });
});
