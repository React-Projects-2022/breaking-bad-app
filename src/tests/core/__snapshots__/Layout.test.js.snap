// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Layout Debe de mostrar <Layout /> correctamente 1`] = `
<Fragment>
  <Header />
  <div
    className="container-md"
  >
    <Routes>
      <Route
        element={<Home />}
        path="/breaking-bad-app"
      />
      <Route
        element={<AllCharacters />}
        path="/breaking-bad-app/all-characters"
      />
      <Route
        element={<Episodes />}
        path="/breaking-bad-app/episodes"
      />
      <Route
        element={<Details />}
        path="/breaking-bad-app/details/:id"
      />
      <Route
        element={<PrincipalCharsVotes />}
        path="/breaking-bad-app/votes"
      />
    </Routes>
  </div>
  <Footer />
</Fragment>
`;
